src/main/java/bio/terra/cli/service/utils/WorkspaceManagerService.java

```diff
@@ -368,21 +367,51 @@ public RoleBindingList getRoles(UUID workspaceId) {
   }
 
   /**
-   * Call the Workspace Manager GET "/api/workspaces/v1/{workspaceId}/resources" endpoint to get a
-   * list of resources (controlled and referenced).
+   * Call the Workspace Manager GET "/api/workspaces/v1/{workspaceId}/resources" endpoint, possibly
+   * multiple times, to get a list of all resources (controlled and referenced) in the workspace.
+   * Throw an exception if the number of resources in the workspace is greater than the specified
+   * limit.
    *
    * @param workspaceId the workspace to query
-   * @param offset the offset to use when listing workspaces (zero to start from the beginning)
-   * @param limit the maximum number of workspaces to return
-   * @param resource the resource type to filter on
-   * @param stewardship the stewardship type to filter on
-   * @return a list of resources, wrapped in a the ResourceList object
+   * @param limit the maximum number of resources to return
+   * @return a list of resources
+   * @throws SystemException if the number of resources in the workspace > the specified limit
    */
-  public ResourceList enumerateResources(
-      UUID workspaceId, int offset, int limit, ResourceType resource, StewardshipType stewardship) {
+  public List<ResourceDescription> enumerateAllResources(UUID workspaceId, int limit) {
     ResourceApi resourceApi = new ResourceApi(apiClient);
     try {
-      return resourceApi.enumerateResources(workspaceId, offset, limit, resource, stewardship);
+      // poll the enumerate endpoint until no results are returned, or we hit the limit
+      List<ResourceDescription> allResources = new ArrayList<>();
+      do {
+        int offset = allResources.size();
+        ResourceList result =
+            resourceApi.enumerateResources(
+                workspaceId, offset, MAX_RESOURCES_PER_ENUMERATE_REQUEST, null, null);
+
+        // add all fetched resources to the running list
+        logger.debug(
+            "Called enumerate endpoints, fetched {} resources", result.getResources().size());
+        allResources.addAll(result.getResources());
+
+        // if we have fetched more than the limit, then throw an exception
+        if (allResources.size() > limit) {
+          throw new SystemException(
+              "Total number of resources ("
+                  + allResources.size()
+                  + ") exceeds the CLI limit ("
+                  + limit
+                  + ")");
+        }
+
+        // if this fetch returned less than the maximum allowed per request, then that indicates
+        // there are no more
+        if (result.getResources().size() < MAX_RESOURCES_PER_ENUMERATE_REQUEST) {
+          break;
+        }
+      } while (true);
```

Comment:
You don't need a do-while if the condition is always `true`; it degenerates to a while loop I think. I can see do while (resultsReturned < maxPerReq) if that's what you originally meant.

Reply:
Changed from `do...while(true)` -> `while(true)`.

Reply:
Yes, fixed the `do...while` loop by moving the `break` condition into the `while`. Thanks!